src/DataMiner/SLNetTypes/SubscriptionFilters/SubscriptionFilter.cs

```diff
@@ -0,0 +1,268 @@
+	///		</description></item>
+	///		<item><description>
+	///			SubscriptionFilterCluster: limits subscription to one particular cluster 
+	///		</description></item>
+	///		<item><description>
+	///			SubscriptionFilterDMA: limits messages to one particular DataMiner Agent
+	///		</description></item>
+	///		<item><description>
+	///			SubscriptionFilterElement: limits messages to one particular element
+	///		</description></item>
+	///		<item><description>
+	///			SubscriptionFilterParameter: limits messages to one particular parameter of a given parameter
+	///		</description></item>
+	/// </list>
+	/// </summary>
+	[Serializable]
+	public class SubscriptionFilter : ICloneable
+	{
+		/// <summary>
+		/// Initializes a new instance of the <see cref="SubscriptionFilter"/> class.
+		/// </summary>
+		public SubscriptionFilter()
+		{
+			Filters = new string[0];
+		}
+
+		/// <summary>
+		/// Initializes a new instance of the <see cref="SubscriptionFilter"/> class.
+		/// </summary>
+		/// <param name="messageType">Name of the message type to subscribe to. All classes inherited from <see cref="DMSMessage"/>
+		/// that are in the <c>SLNetTypes.dll</c> assembly can be used. The <c>messageType</c> name 
+		/// is the fully qualified name, starting from <c>Skyline.DataMiner.Net</c>
+		/// (e.g. "<c>AlarmEventMessage</c>" subscribes to all <see cref="AlarmEventMessage"/> messages)
+		/// </param>
+		public SubscriptionFilter(string messageType) : this(messageType, null)
+		{
+
+		}
+
+		/// <summary>
+		/// Initializes a new instance of the <see cref="SubscriptionFilter"/> class.
+		/// </summary>
+		/// <param name="messageType">Type of the message.</param>
+		/// <param name="filters">The filters.</param>
+		/// <exception cref="System.ArgumentNullException"><paramref name="messageType"/> is <see langword="null"/>.</exception>
+		public SubscriptionFilter(string messageType, string[] filters)
+		{
+		}
+
+		/// <summary>
+		/// Initializes a new <see cref="SubscriptionFilter"/> instance for the given message type.
+		/// </summary>
+		/// <param name="messageType">Message type.</param>
+		public SubscriptionFilter(Type messageType) : this(messageType, SubscriptionFilterOptions.None)
+		{
+		}
+
+		/// <summary>
+		/// Initializes a new instance of the <see cref="SubscriptionFilter"/> class.
+		/// </summary>
+		/// <param name="messageType">Type of the message.</param>
+		/// <param name="options">The options.</param>
+		/// <exception cref="System.ArgumentNullException"><paramref name="messageType"/> is <see langword="null"/>.</exception>
+		public SubscriptionFilter(Type messageType, SubscriptionFilterOptions options) : this()
+		{
+		}
+
+		protected SubscriptionFilter(SubscriptionFilter subscriptionFilter)
+			: this(subscriptionFilter.MessageType, subscriptionFilter.Filters)
+		{
+		}
+
+		/// <summary>
+		/// The message type about which the <see cref="SubscriptionFilter"/> handles.
```

Comment:
This isn't quite clear. Could you rephrase it?

Reply:
updated